src/g3log/active.hpp

```diff
@@ -57,9 +72,10 @@ namespace kjellkod {
       }
 
       /// Factory: safe construction of object before thread start
-      static std::unique_ptr<Active> createActive() {
+      static std::unique_ptr<Active> createActive(
+         const native_thread_name &threadNamePrefix = DefaultThreadNamePrefix) {
          std::unique_ptr<Active> aPtr(new Active());
-         aPtr->thd_ = std::thread(&Active::run, aPtr.get());
+         aPtr->thd_ = std::thread(&Active::run, aPtr.get(), threadNamePrefix);
```

Comment:
I think this concept is interesting but maybe a bit deceiving. While you can change the name, isn't this always just the thread for the active object running in the background? 

Not sure what your purpose is with this. Is it to connect each active object with the sink from a debugging perspectivie? 

Please explain the problem you are trying to solve and how this helps you and others.

Reply:
I do see the power of this overall but maybe from a different perspective that I need to think about. 

Consider if you could call to set your thread-name for the `LOG` calling thread? 
We already capture the thread_id, and it's optional to include that in the formatting output of the logs. 

I know people have requested an easier way to see which LOG call comes from which thread, and of course a thread_ID, even if it's usable isn't super easy to parse for human eyes compared to plain text. 

I'm not saying you should change the PR towards this. I'm only musing on the power of this functionality and how it could be applied differently. 

Looking forward to your reply about the purpose of the PR

Reply:
General idea was to name threads from g3log, as there may be a lot of threads in the app and it is easier to debug them when you have descriptive names on the table.

Anyway, your idea looks good, I'll think about.